store/src/java/com/zimbra/cs/mailbox/AutoSendDraftTask.java

```
@@ -73,7 +114,35 @@ public class AutoSendDraftTask extends ScheduledTask<Object> {
         mailSender.setOriginalMessageId(StringUtil.isNullOrEmpty(msg.getDraftOrigId()) ? null : new ItemId(msg.getDraftOrigId(), mbox.getAccountId()));
         mailSender.setReplyType(StringUtil.isNullOrEmpty(msg.getDraftReplyType()) ? null : msg.getDraftReplyType());
         mailSender.setIdentity(StringUtil.isNullOrEmpty(msg.getDraftIdentityId()) ? null : mbox.getAccount().getIdentityById(msg.getDraftIdentityId()));
-        mailSender.sendMimeMessage(new OperationContext(mbox), mbox, msg.getMimeMessage());
+        Provisioning provisioning = Provisioning.getInstance();
+        Account delegatorAccount = provisioning.getAccountById(getDelegatorAccountId());
+        if (delegatorAccount != null && Provisioning.onLocalServer(delegatorAccount)) {
+            delegatorMbox = MailboxManager.getInstance().getMailboxByAccount(delegatorAccount);
+            /* if delegator is on same server, just fetch the delegatorMbox object
+            and use it for sending the Mime. */
+            mailSender.sendMimeMessage(new OperationContext(mbox), delegatorMbox, msg.getMimeMessage());
+        } else {
+            /* if delegator is on different server, invoke the saveDraft request on that server. */
+            Element rqstforDelegatorAcnt = getRequest();
+            AuthToken tokenForSetup = null;
+            if (delegatorAccount != null) {
+                try {
+                    tokenForSetup = AuthProvider.getAuthToken(delegatorAccount, AuthToken.Usage.AUTH);
+                } catch (AuthProviderException e) {
+                    throw new RuntimeException(e);
```

Please throw an appropriate exception instead of RuntimeException